pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -298,7 +298,7 @@ def _save_model(self, filepath: str, trainer, pl_module):
 
     def check_monitor_top_k(self, current) -> bool:
         less_than_k_models = len(self.best_k_models) < self.save_top_k
```

Comment:
i'd suggest returning True try right away if self.save_top_k == -1

Reply:
hmm, I'm wondering if the warning below is in the right place, since we return early if we don't have k models.